shopinvader_sale_coupon/services/customer.py

```diff
@@ -12,35 +12,52 @@ def _parser_coupon(self):
         return [
             "id",
             "code",
-            "state",
             "expiration_date",
             (
                 "program_id:program",
                 [
                     "id",
                     "name",
-                    "reward_type",
-                    "reward_product_id",
-                    "reward_product_quantity",
-                    "discount_type",
-                    "discount_percentage",
-                    "discount_fixed_amount",
+                    "program_type",
+                    ("reward_ids:rewards", self._parser_coupon_reward()),
                 ],
             ),
         ]
 
+    def _parser_coupon_reward(self):
+        return [
+            "id",
+            "reward_type",
+            "discount",
+            "discount_mode",
+            "discount_applicability",
+            ("all_discount_product_ids:discount_products", self._parser_product()),
+            ("reward_product_id:reward_product", self._parser_product()),
+        ]
```

Comment:
Here I'm defining `_parser_product` but I guess the best would be to use https://github.com/akretion/odoo-shopinvader/blob/6d3ed9e88adb1f6e718184dba1fd37506a690de8/shopinvader_restapi/services/abstract_sale.py#L46

The problem is that it is not defined on the same service component...